keps/sig-storage/20190709-csi-snapshot.md

```diff
@@ -0,0 +1,102 @@
+---
+title: CSI Snapshot 
+authors:
+  - "@jingxu97"
+  - "@xing-yang"
+owning-sig: sig-storage
+participating-sigs:
+  - sig-storage
+reviewers:
+  - "@msau42"
+  - "@saad-ali"
+  - "@thockin"
+approvers:
+  - "@msau42"
+  - "@saad-ali"
+  - "@thockin"
+editor: TBD
+creation-date: 2019-07-09
+last-updated: 2019-07-29
+status: implementable
+see-also:
+  - n/a
+replaces:
+  - n/a
+superseded-by:
+  - n/a
+---
+
+# Title
+
+CSI Snapshot
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Test Plan](#test-plan)
+  - [Unit tests](#unit-tests)
+  - [E2E tests](#e2e-tests)
+- [Graduation Criteria](#graduation-criteria)
+  - [Alpha-&gt;Beta](#alpha-beta)
+  - [Beta-&gt;GA](#beta-ga)
+- [Changes](#changes)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Summary
+
+This KEP is written after the original design doc has been approved and implemented. Design for CSI Volume Snapshot Support in Kubernetes is incorporated as part of the [CSI Volume Snapshot in Kubernetes Design Doc](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/csi-snapshot.md).
+
+The rest of the document includes required information missing from the original design document: test plan and graduation criteria.
+
+## Test Plan
+
+### Unit tests
+
+* Unit tests around snapshot creation and deletion logic.
+* Unit tests around VolumeSnapshot and VolumeSnapshotContent binding logic.
+* Unit tests for creating volume from snapshot.
+
+### E2E tests
+
+* e2e tests for creating/deleting snapshot.
+* e2e tests for creating volume from snapshot.
+
+## Graduation Criteria
+
+### Alpha->Beta
+
+* Feature complete, including:
+  * Create/delete volume snapshots
+  * Create new volumes from a snapshot
+  * SnapshotContent Deletion/Retain Policy
+  * Snapshot Object in Use Protection
+* Unit and e2e tests implemented
+
+### Beta->GA
+
+* Snapshot feature is used as a basic building block in other advanced applications. 
+* Feature deployed in production and have gone through at least one K8s upgrade.
+
+## Changes
```

Comment:
Can we separate this list between what is already complete and what is in progress for beta so it is easier to understand current state vs what is being done.

Reply:
Sure